dashboard/app/models/backpack.rb

```diff
@@ -13,4 +13,23 @@
 #  index_backpacks_on_user_id  (user_id)
 #
 class Backpack < ApplicationRecord
+  belongs_to :user
+  validates_uniqueness_of :storage_app_id
```

Comment:
Do we need an index on storage_app_id?

Reply:
We will only need to look up by user, not storage_app_id

Reply:
I'm not sure but won't Rails query the table for `storage_app_id` when it save a backpack to verify uniqueness?

Reply:
ah good point. Looking at the [docs](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) for `uniqueness` and it seems that doing this doesn't even always guarantee uniqueness. I'll remove the uniqueness constraints and fix up the indexes as a follow-up.